orbit/pkg/update/flag_runner.go

```diff
@@ -126,6 +126,180 @@ func (r *FlagRunner) DoFlagsUpdate() (bool, error) {
+		stat, err := os.Stat(extensionAutoLoadFile)
+		if errors.Is(err, os.ErrNotExist) {
+			log.Debug().Msg(extensionAutoLoadFile + " not found, nothing to update")
+			// we do not want orbit to restart
+			return false, nil
+		}
+
+		if stat.Size() > 0 {
+			// handle case 2: create/truncate the extensions.load file and let the runner interrupt, so that
+			// osquery can't startup without the extensions that were previously loaded
+			// WriteFile will create the file if it doesn't exist, and it handles Close for us
+			err := os.WriteFile(extensionAutoLoadFile, []byte(""), constant.DefaultFileMode)
+			if err != nil {
+				// we do not want orbit to restart
+				return false, fmt.Errorf("extensionsUpdate: error creating file %s, %w", extensionAutoLoadFile, err)
+			}
+			// we want to return true here, and restart with the empty extensions.load file
+			// so that we "unload" the previously loaded extensions
+			return true, nil
+		}
+		// we do not want orbit to restart
+		return false, nil
+	}
+
+	type ExtensionInfo struct {
+		Platform string `json:"platform"`
+		Channel  string `json:"channel"`
+	}
+
+	var data map[string]ExtensionInfo
+	err = json.Unmarshal(config.Extensions, &data)
+	if err != nil {
+		// we do not want orbit to restart
+		return false, fmt.Errorf("error unmarshing json extensions config from fleet: %w", err)
+	}
+
+	var sb strings.Builder
+	for extensionName, extensionInfo := range data {
+		// infer filename from extension name
+		// osquery enforces .ext, so we just add that
+		// we expect filename to match extension name
+		filename := extensionName + ".ext"
+
+		// we don't want path traversal and the like in the filename
+		if strings.Contains(filename, "..") || strings.Contains(filename, "/") {
+			log.Info().Msgf("invalid characters found in filename (%s) for extension (%s): skipping", filename, extensionName)
+			continue
+		}
+
+		// add "extensions/" as a prefix to the targetName, since that's the namespace we expect for extensions on TUF
+		targetName := "extensions/" + extensionName
+		platform := extensionInfo.Platform
+		channel := extensionInfo.Channel
+
+		rootDir := r.updateRunner.updater.opt.RootDirectory
+
+		// update our view of targets
+		r.updateRunner.UpdateRunnerOptTargets(targetName)
+		r.updateRunner.updater.SetExtensionsTargetInfo(targetName, platform, channel, filename)
+
+		// the full path to where the extension would be on disk, for e.g. for extension name "hello_world"
+		// the path is: <root-dir>/bin/extensions/hello_world/<platform>/<channel>/hello_world.ext
+		path := filepath.Join(rootDir, "bin", "extensions", extensionName, platform, channel, filename)
+
+		meta, err := r.updateRunner.updater.Lookup(targetName)
+		if err != nil {
+			// we do not want orbit to restart
+			return false, fmt.Errorf("unable to lookup metadata for target: %s, %w", targetName, err)
+		}
+
+		_, localHash, err := fileHashes(meta, path)
```

Comment:
This will always return `os.ErrNotExists` on the first addition of the extension to the `r.updateRunner`. (Beware that the development `main.sh` script always runs with a `10s` update interval, versus the default/production value of `1h`.)

Reply:
This is expected, and I think ok as well. We do call `UpdateAction()` after successfully fetching extensions from `/config`.  

This is more of an optimization, since `UpdateAction` will compare local and meta hashes, and we want to provide the local hash, otherwise it will redownload the extension and restart.

The longer update interval, would mean longer waiting (and I think default is 15 minutes, or probably that's just the hard-coded fallback value..?). 

Thoughts?

Reply:
My mistake, correct, it's 15m.

--

Let's clarify that the only time `DoExtensionConfigUpdate` returns `true, nil` is when the Fleet server has cleared extensions and the host has some extensions set up.

Either document with named returns as below, or add the above sentence to the method doc.
`func (r *ExtensionRunner) DoExtensionConfigUpdate() (extensionsCleared bool, err error) {`

On the caller side this should actually be:
```go
extensionsCleared, err := r.DoExtensionConfigUpdate()
if err != nil {
  log.Info().Err(err).Msg("ext update failed")
}
if extensionsCleared {
  log.Info().Msg("extensions were cleared on the server")
  return nil
}
```
--

For the refactor issue later:
`DoExtensionConfigUpdate` is too long (thus hard to follow/review). Let's refactor it using multiple functions.